### PR TITLE
Resolve audit fails due to nil pointer reference to options.Csv

### PR DESCRIPTION
**Pull request by yashoza19**
Signed-off-by: yoza <yoza@redhat.com>

## Description of PR
Opcap audits were failing since a nil pointer was passed to the option.Csv from resultCSV.

Fixes #263 
Fixes #265 
Fixes #266 
Fixes #267 

## Changes
Modified options.Csv to be a pointer

## Checklist (required)
- [x] I have reviewed and followed the [contribution guidelines](https://github.com/opdev/opcap/blob/main/docs/contribution.md)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if needed
- [x] I have checked that my changes pass all tests

### CODE REVIEW

**Review by bcrochet — APPROVED** [review]
/lgtm

Looks good. Might be a good idea to add some tests to the report tests? Make sure that a nil CSV is handled?

File: internal/capability/operand_install.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func operandInstall(ctx context.Context, opts ...auditOption) auditFn {
 		if err != nil {
 			return fmt.Errorf("could not get CSV: %v", err)
 		}
+		options.Csv = csv
```
Comment (bcrochet) [inline internal/capability/operand_install.go:70]: Need to check for nil here as well.

File: internal/capability/operator_install.go [hunk 2]
```diff
@@ -56,9 +56,9 @@ func operatorInstall(ctx context.Context, opts ...auditOption) auditFn {
 				return err
 			}
 		}
-		options.Csv = *resultCSV
+		options.Csv = resultCSV
 		if resultCSV == nil {
-			options.Csv = operatorv1alpha1.ClusterServiceVersion{}
+			options.Csv = &operatorv1alpha1.ClusterServiceVersion{}
```
Comment (bcrochet) [inline internal/capability/operator_install.go:61]: What's more useful to us? A zero value for CSV? Or a nil? I feel like a nil would convey better information to the reporting functions.

Comment (yashoza19) [reply internal/capability/operator_install.go:61]: I agree with having a nil value for CSV for now.

File: internal/capability/operand_install.go (hunk 1 above)
Comment (yashoza19) [reply internal/capability/operand_install.go:71]: Done.

File: internal/capability/operator_install.go (hunk 2 above)
Comment (yashoza19) [reply internal/capability/operator_install.go:61]: I think it cannot be nil, since it is required by the report. It errors out on the report when submitting a nil value.

File: internal/capability/report.go [hunk 3]
```diff
@@ -21,8 +21,8 @@ Channel: {{ .Subscription.Channel }}
 Catalog Source: {{ .Subscription.CatalogSource }}
 Install Mode: {{ .Subscription.InstallModeType }}
 Result: {{ if .CsvTimeout }}timeout{{ else }}{{ .Csv.Status.Phase }}{{ end }}
-Message: {{ .Csv.Status.Message }}
-Reason: {{ .Csv.Status.Reason }}
+Message: {{if .Csv}}{{ .Csv.Status.Message }}{{else}}CSV installation has failed due to a timeout{{end}}
+Reason: {{if .Csv}}{{ .Csv.Status.Reason }}{{else}}CSV installation has failed due to a timeout{{end}}
 -----------------------------------------
 `
 	operatorJsonReportTemplate = `{"level":"info","message":"{{ if .CsvTimeout }}timeout{{ else }}{{ .Csv.Status.Phase }}{{ end }}","package":"{{ .Subscription.Package }}","channel":"{{ .Subscription.Channel }}","installmode":"{{ .Subscription.InstallModeType }}"}`
```
Comment (bcrochet) [inline internal/capability/report.go:28]: You'll need to wrap the .Csv.Status.Phase here since it can be nil

File: internal/capability/report.go [hunk 4]
```diff
@@ -21,8 +21,8 @@ Channel: {{ .Subscription.Channel }}
 Catalog Source: {{ .Subscription.CatalogSource }}
 Install Mode: {{ .Subscription.InstallModeType }}
 Result: {{ if .CsvTimeout }}timeout{{ else }}{{ .Csv.Status.Phase }}{{ end }}
```
Comment (bcrochet) [inline internal/capability/report.go:23]: This should also be wrapped.

**Review by bcrochet — APPROVED** [review]
/lgtm
